broker-daemon/block-order-worker/index.js

```diff
@@ -340,21 +342,38 @@ class BlockOrderWorker extends EventEmitter {
 
     this.logger.info('Creating order for BlockOrder', { blockOrderId: blockOrder.id })
 
-    const order = await OrderStateMachine.create(
+    const osm = await OrderStateMachine.create(
       {
         relayer,
         engines,
         logger,
-        store,
-        onRejection: (err) => {
-          this.failBlockOrder(blockOrder.id, err)
-        }
+        store
       },
       blockOrder.id,
       { side, baseSymbol, counterSymbol, baseAmount, counterAmount }
     )
 
-    this.logger.info('Created order for BlockOrder', { blockOrderId: blockOrder.id, orderId: order.orderId })
+    // We are hooking into the complete lifecycle event of an order state machine to trigger
+    // the completion of a blockorder
+    osm.once('complete', async () => {
+      this.completeBlockOrder(blockOrder.id)
+        .catch(e => {
+          this.logger.error(`BlockOrder failed to be completed from order`, { id: blockOrder.id, error: e.stack })
+        })
+        .then(() => osm.removeAllListeners())
```

Comment:
if we are using a `once`, do we need to clean up listeners here?

Reply:
We still need to clean up, because we would have a `reject` listener still open on the `osm`